### PR TITLE
Implemented feature to fetch product current stock

### PR DESCRIPTION
**Pull request by Neche-Stephen**
I Implemented feature to fetch product current stock

Screenshot 2024-07-24 at 8 37 19 PM

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -164,7 +164,22 @@
 				<groupId>org.springframework.boot</groupId>
 				<artifactId>spring-boot-maven-plugin</artifactId>
 			</plugin>
+
+			<plugin>
+				<groupId>org.flywaydb</groupId>
```
Comment (Fifetoyi) [inline pom.xml:169]: we already have a flyway dependency in the pom.xml

File: src/main/java/hng_java_boilerplate/config/WebSecurityConfig.java [hunk 2]
```diff
@@ -73,7 +73,7 @@ public SecurityFilterChain httpSecurity(HttpSecurity httpSecurity) throws Except
                         httpRequests
 //                                .requestMatchers("").hasAuthority(String.valueOf(Role.ADMIN))
                                 .requestMatchers("/", "/v3/api-docs", "/swagger-ui/index.html" ,"/api/v1/auth/**").permitAll()
-                                .requestMatchers("/api/v1/auth/logout","/api/**").authenticated())
+                                .requestMatchers("/api/v1/auth/logout","/api/**", "/api/v1/products/**").authenticated())
```
Comment (Fifetoyi) [inline src/main/java/hng_java_boilerplate/config/WebSecurityConfig.java:76]: not necessary

File: src/main/java/hng_java_boilerplate/product/service/ProductService.java [hunk 3]
```diff
@@ -10,4 +10,6 @@ public interface ProductService {
 
     //Method to Search for products with certain criteria, returns a list of products
```
Comment (Fifetoyi) [inline src/main/java/hng_java_boilerplate/product/service/ProductService.java:11]: remove comments

File: src/main/java/hng_java_boilerplate/config/WebSecurityConfig.java (hunk 2 above)
Comment (Neche-Stephen) [reply src/main/java/hng_java_boilerplate/config/WebSecurityConfig.java:76]: Alright. I will update it.

File: src/main/java/hng_java_boilerplate/product/service/ProductService.java (hunk 3 above)
Comment (Neche-Stephen) [reply src/main/java/hng_java_boilerplate/product/service/ProductService.java:11]: Noted.

File: pom.xml (hunk 1 above)
Comment (Neche-Stephen) [reply pom.xml:169]: Kindly check and confirm if there is a  flyway dependency in the pom.xml, When I cloned it was not present, I checked again now and it is still not present.

Screenshot 2024-07-24 at 8 59 06 PM

Comment (Neche-Stephen) [reply pom.xml:169]: Kindly check and confirm if there is a  flyway dependency in the pom.xml, When I cloned it was not present, I checked again now and it is still not present.

Screenshot 2024-07-24 at 8 59 06 PM

**Pull request closed without merging**